### PR TITLE
Update environment variables doc

### LINKED ISSUE #5746: `Environment variables` doc should mention XDG_* variables
I think uv uses
- XDG_CONFIG_HOME
- XDG_BIN_HOME
- XDG_DATA_HOME
- XDG_CACHE_HOME

these should be mentioned in `Environement varibles` page.

### PR DESCRIPTION
**Pull request by j178**
## Summary

This PR adds descriptions for the `UV_TOOL_DIR`, `UV_PYTHON_INSTALL_DIR` and `XDG_*` environment variables.

Additionally, it moves some env vars that are not command-line arguments to the below "uv respects" section.

Closes #5746

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -628,11 +637,12 @@ uv accepts the following command-line arguments as environment variables:
...
-In each case, the corresponding command-line argument takes precedence over an environment variable.
-
-In addition, uv respects the following environment variables:
-
+- `XDG_CONFIG_HOME`: Used to specify the path to uv user-level configuration directory on Linux
+  _and_ macOS systems.
+- `XDG_CACHE_HOME`: Used to specify the directory where uv stores cache files on Linux system.
+- `XDG_DATA_HOME`: Used to specify the directory where uv stores managed Python installations and
+  managed tools on Linux system.
+- `XDG_BIN_HOME`: Used to specify the directory where executables are installed into.
```
Comment (j178) [inline README.md:645]: should be updated after https://github.com/astral-sh/uv/pull/5806/

File: README.md
Comment (j178) [inline README.md:1]: Should we remove the `environment variables` section from the README and link it to the documentation instead?

Comment (zanieb) [reply README.md:1]: I'll do that soon!

**Review by zanieb — APPROVED** [review]
Thank you!

**Pull request merged**